src/api/views/licence_plate_image_view.py

```diff
@@ -0,0 +1,81 @@
+import os
+import re
+import shutil
+
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.permissions import AllowAny
+
+from src.core.views import BackendResponse, _OriginAPIView
+from src.api.models import Image, LicencePlate
+
+from anpr.license_plate_recognition import ANPR
+from anpr.google_vision_ocr import GoogleVisionOCR
+
+
+class LicencePlateImageView(_OriginAPIView):
+    """
+    A view class to handle the incoming images in base64-format. From within this view, a image
+    processing is performed and a request is sent to the Google Vision API. The image itself
+    will NOT be stored in the database, but deleted when the function ends.
+    """
+
+    permission_classes = [AllowAny]
+    origins = ["rpi"]
+
+    def post(self, request: Request, format=None) -> BackendResponse:
+        if (resp := super().post(request, format)) is not None:
+            return resp
+        file = request.data["file"]  # type: ignore
+        garage_id: int = request.headers["PO3-GARAGE-ID"]
+        Image.objects.create(image=file)
+        lp = ""
+        try:
+            lp = perform_ocr(file.name)  # type: ignore
+        except Exception as e:
+            delete_file()
+            return BackendResponse(
+                [f"An error occurred while parsing the licence plate image: {e}."],
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+        out_int = LicencePlate.handle_licence_plate(strip_special_chars(lp), garage_id)
+        delete_file()
+        return BackendResponse(
+            {"response": f"Successfully registered licence plate {lp}."}
+            if out_int == 1
+            else {"response": f"Successfully signed out licence plate {lp}."},
+            status=status.HTTP_200_OK,
+        )
+
+
+def perform_ocr(image_path: str) -> str:
+    anpr = ANPR(None, GoogleVisionOCR(), formats=["N-LLL-NNN"], verbosity=0)  # type: ignore
+    ocr_results = anpr.find_and_ocr(
+        os.path.join(os.getcwd(), f"src/media/images/{image_path}"),
+        doSelection=False,
+    )
+    lp = "".join([lp.text for lp in ocr_results])
```

Comment:
ocr_results is a list of the different detected licence plates (for example if there are multiple licence plates in one image). I think joining them will create an invalid licence plate

Reply:
But when do we have multiple licence plates on one image?

Reply:
I added possibility in case multiple cars enter the garage at the same time and the camera can see both licence plates. But this will indeed only happen if the camera is high enough. Maybe it's safer to use the first element in the list instead of the combined licence plates.